pysb/core.py

```diff
@@ -1671,6 +1689,32 @@ def rename(self, c, new_name):
             del m[c.name]
 
 
+class OdeView(collections.Sequence):
+    """Compatibility shim for the Model.odes property."""
+
+    # This is necessarily coupled pretty tightly with Model. Note that we
+    # faithfully emulate the detail of the original implementation in which odes
+    # is an empty list before the equation generation process is run (and after
+    # reset_equations is called). Now the "empty" condition is when species is
+    # empty.
+
+    def __init__(self, model):
+        self.model = model
+
+    def __getitem__(self, key):
+        if not self.model.species:
+            raise IndexError('list index out of range')
+        if isinstance(key, slice):
+            return [self[k] for k in range(*key.indices(len(self)))]
+        sr = self.model.stoichiometry_matrix[key]
+        terms = [self.model.reactions[i]['rate'] * v
+                 for i, v in zip(sr.indices, sr.data)]
```

Comment:
Replacing this statement with
```python
terms = [sympy.Mul(self.model.reactions[i]['rate'], v, evaluate=False)
         for i, v in zip(sr.indices, sr.data)]
```
speeds the `time_egfr_equations_max_iter_8` benchmark up by around 10% for me.